### PR TITLE
Rely on lockfile for `alr setenv`

### PR DESCRIPTION
**Pull request by mosteo**
This way externals are not attempted to be detected every time.

Only the last commit is new (this was originally part of PR #355, so it has already been reviewed).

### CODE REVIEW

**Comment by mosteo** [comment]
I've moved the `workflow/setenv` test to `setenv/basic`, and added a `setenv/with-external` that tests the output when there are external dependencies.

File: src/alr/alr-commands-setenv.adb [hunk 1]
```diff
@@ -14,16 +14,7 @@ package body Alr.Commands.Setenv is
...
-      --  would prevent directly sourcing the output. TODO: remove this once
-      --  the lockfile is used, that will make this unnecessary.
-      declare
-         Old_Level : constant Simple_Logging.Levels := Alire.Log_Level;
-      begin
-         Alire.Log_Level := Simple_Logging.Always;
-         Alr.Build_Env.Print (Alr.Root.Current);
-         Alire.Log_Level := Old_Level;
-      end;
+      Alr.Build_Env.Print (Alr.Root.Current);
```
Comment (Fabien-Chouteau) [inline src/alr/alr-commands-setenv.adb:17]: Why are you removing the temporary change of output level?

Comment (mosteo) [reply src/alr/alr-commands-setenv.adb:17]: After #385 the problem with the logging no longer happens. The detection status either goes away on regular runs or doesn't happen on redirections.

Comment (Fabien-Chouteau) [reply src/alr/alr-commands-setenv.adb:17]: ok :+1:

**Review by Fabien-Chouteau — APPROVED** [review]

**Pull request merged**